---
title: "Annotation for timepoint2"
author: "Ninghui"
date: "2025-07-11"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r cars}
library(Biostrings)
```

```{r}
# Read the Protein FASTA file
setwd(""input-dir"")
fasta_file<-readAAStringSet("P.tabuliformis_V1.0.Protein.fa")
```

```{r}
# View the protein file
head(fasta_file)
```

```{r}
# Convert the protein file to csv format
fasta_df <- data.frame(name = names(fasta_file), sequence = as.character(fasta_file))
write.csv(fasta_df, "protein_sequences.csv", row.names = FALSE)
```

```{r}
# Read the significant differentially expressed genes for tp2
setwd(""input-dir"")
deg_ids_tp2<-read.csv("deg2_tp2 (significant genes after tidyverse).csv")$row
```

```{r}
# Extract Main Isoforms (".1" Transcripts)
main_isoforms <- fasta_file[grepl("\\.1$", names(fasta_file))]
head(main_isoforms)
```
```{r}
# Removes the `.1` suffix from sequence names to match the gene IDs from the DEG file.
names(main_isoforms) <- sub("\\.1$", "", names(main_isoforms))
head(main_isoforms)
```

```{r}
# Extract the correlated protein sequence 
deg_fasta_tp2<-main_isoforms[names(main_isoforms)%in%deg_ids_tp2]
head(deg_fasta_tp2)
```
```{r}
# Write the DEGs protein sequences to a new FASTA file for BLAST
writeXStringSet(deg_fasta_tp2, filepath = "deg_tp2_proteins.fa")
```

```{r}
# Select the DEGs that have upregulated or downregulated functions
selected_ids <- c("Pt1G72030", "Pt1G04990", "Pt1G05120", 
                  "Pt1G05390", "Pt1G27830", "Pt1G65140", "Pt1G69580")
```

```{r}
# Extract correlated protein sequences for these upregulated or downregulated genes
selected_seqs <- main_isoforms[names(main_isoforms) %in% selected_ids]
```

```{r}
selected_seqs
```
```{r}
# Read the file includes these upregulated or downregulated genes
setwd("input-dir")
sig_info<-read.csv("up_and_down_genes_tp2.csv")
```

```{r}
# Just keep the gene ID and significant information
library(dplyr)
selected_info <- sig_info %>%
  filter(row %in% selected_ids) %>%
  select(row, signif)  
```

```{r}
# Check the gene ID and significant information
head(selected_info)
```
```{r}
selected_info
```

```{r}
# Extract the significant information
status_vector <- selected_info$signif
```

```{r}
names(status_vector) <- selected_info$row
```

```{r}
selected_seqs <- main_isoforms[names(main_isoforms) %in% selected_ids]
```

```{r}
# Extracted homologous protein sequence which has the same functions with DEGs (upregulated or downregulated functions)
names(selected_seqs) <- paste0(names(selected_seqs), "_", status_vector[names(selected_seqs)])
```

```{r}
# Check these protein sequences with functions
selected_seqs
```
